### PR TITLE
Fix shim for LTSS-ERICSSON

### PR DESCRIPTION
**Pull request by dzedro**
- Related ticket: https://progress.opensuse.org/issues/128993
- Verification run:
https://dzedro.suse.cz/tests/1157
https://openqa.suse.de/tests/11075329

### CODE REVIEW

File: lib/qam.pm [hunk 1]
```diff
@@ -81,6 +81,28 @@ sub add_test_repositories {
...
+    if (is_sle('=12-SP3')) {
+        my $arch = get_var('ARCH');
+        my $url = "http://dist.suse.de/ibs/SUSE/Updates/SLE-SERVER/12-SP3-LTSS-TERADATA/$arch/update/";
+        # don't add repo when it's already present
+        unless ($system_repos =~ /$url/) {
+            zypper_call("--no-gpg-checks ar -f $gpg $url '12-SP3-LTSS-TERADATA-Updates'");
+        }
+    }
+    # shim update will fail with old grub2 due to old signature
+    if (check_var('MACHINE', 'uefi')) {
```
Comment (rfan1) [inline lib/qam.pm:102]: How about:
if (get_var('MACHINE') =~ /uefi/)

Comment (rfan1) [reply lib/qam.pm:102]: We have other machine types like `uefi-2G`

Comment (dzedro) [reply lib/qam.pm:102]: I guess I will change to proposal or to look for `UEFI`.

**Comment by dzedro** [comment]
I rerun the test, it failed because the update is already released, but [UEFI part](https://openqa.suse.de/tests/11131067#step/install_update/74) was executed as expected.

**Comment by rfan1** [comment]
Hello @dzedro https://progress.opensuse.org/issues/129361, seems your changes impact tests on aarch64 platform.

**Comment by mloviska** [comment]
Also sle-micro is impacted
https://openqa.suse.de/tests/overview?distri=sle-micro&version=5.1&build=20230515-1&groupid=420

**Comment by mloviska** [comment]
> Can we quickly revert this and re-work it again?

there is a fix already out there https://github.com/os-autoinst/os-autoinst-distri-opensuse/pull/17078

**Pull request merged**